Realm/RLMRealm.mm

```diff
@@ -792,7 +795,7 @@ + (void)setDefaultRealmSchemaVersion:(uint64_t)version withMigrationBlock:(RLMMi
 }
 
 + (void)setSchemaVersion:(uint64_t)version forRealmAtPath:(NSString *)realmPath withMigrationBlock:(RLMMigrationBlock)block {
-    if (RLMGetAnyCachedRealmForPath(realmPath)) {
+    if (RLMGetAnyCachedRealmForPath(realmPath) && schemaVersionForPath(realmPath) != version) {
```

Comment:
This check needs to be within the `@synchronized` block below to avoid a race condition (that'd require really weird code to hit).